From 7ffa1f91350d51cbd26a5ee1ac4aada7ae1ca7be Mon Sep 17 00:00:00 2001
From: Adam Bookatz <bookatz@google.com>
Date: Wed, 11 Dec 2024 17:22:46 -0800
Subject: [PATCH] startActivityForResult with earlier new Intent

We already make sure to use a copy of the Intent, but now we do so
earlier. See bug.

Bug: 353680402
Flag: EXEMPT bugfix
Test: manual
Test: atest com.android.settings.users.UserSettingsTest
            com.android.settings.users.UserDetailsSettingsTest
Change-Id: I860e9e606de6b8d3c99fa52a63b72ba7a99ce179
Merged-In: I860e9e606de6b8d3c99fa52a63b72ba7a99ce179
(cherry picked from commit b7240e2f0c50455a1c8f3ae1fc4f27d55b86e89b)
---
 src/com/android/settings/users/AppRestrictionsFragment.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/users/AppRestrictionsFragment.java b/src/com/android/settings/users/AppRestrictionsFragment.java
index 379e4bbaec..60b7925967 100644
--- a/src/com/android/settings/users/AppRestrictionsFragment.java
+++ b/src/com/android/settings/users/AppRestrictionsFragment.java
@@ -669,8 +669,9 @@ public class AppRestrictionsFragment extends SettingsPreferenceFragment implemen
          */
         private Intent assertSafeToStartCustomActivity(Intent intent) {
             EventLog.writeEvent(0x534e4554, "223578534", -1 /* UID */, "");
+            final Intent vettedIntent = new Intent(intent);
             ResolveInfo resolveInfo = mPackageManager.resolveActivity(
-                    intent, PackageManager.MATCH_DEFAULT_ONLY);
+                    vettedIntent, PackageManager.MATCH_DEFAULT_ONLY);
 
             if (resolveInfo == null) {
                 throw new ActivityNotFoundException("No result for resolving " + intent);
@@ -684,7 +685,6 @@ public class AppRestrictionsFragment extends SettingsPreferenceFragment implemen
 
             // We were able to vet the given intent this time. Make a copy using the components
             // that were used to do the vetting, since that's as much as we've verified is safe.
-            final Intent vettedIntent = new Intent(intent);
             vettedIntent.setComponent(activityInfo.getComponentName());
             vettedIntent.setPackage(activityInfo.packageName);
             return vettedIntent;
-- 
2.50.1.351.gcf940e82a1-goog

